File: android/app/src/main/java/com/lightningkite/khrysalistemplate/layouts/PongDemoXml.kt
//
// PongDemoXml.swift
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysalistemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalistemplate.R

class PongDemoXml {

    lateinit var customView: com.lightningkite.khrysalis.views.CustomView
    lateinit var customViewDelegate: com.lightningkite.khrysalistemplate.vg.PongDelegate
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.pong_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        customView = view.findViewById<com.lightningkite.khrysalis.views.CustomView>(R.id.customView)
        customViewDelegate = view.findViewById<CustomView>(R.id.customView).delegate as com.lightningkite.khrysalistemplate.vg.PongDelegate
        
        return view
    }
}

File: ios/ButterflyTemplate/swiftResources/layouts/PongDemoXml.swift
//
// PongDemoXml.swift
// Created by Khrysalis XML
//

import UIKit
import LKButterfly
import MapKit

public class PongDemoXml {
    
    public unowned var xmlRoot: UIView!
    private var _layoutTests: Array<()->Bool> = []
    private func pickLayout(test: @escaping()->Bool) -> Bool {
        _layoutTests.append(test)
        return test()
    }
    public func setup(dependency: ViewControllerAccess) -> UIView {
        let view = FrameLayout(frame: .zero)
        view.padding = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
        view.addSubview(
            CustomView(frame: .zero),
            minimumSize: CGSize(width: 0, height: 0),
            size: CGSize(width: -1, height: -1),
            margin: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
            padding: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
            gravity: .centerFill
        ) { view in 
            self.customView = view
            applyColor(view, R.color.black) { (c, s) in
                view.backgroundColor = c
            }
            let dg = PongDelegate()
            view.delegate = dg
            self.customViewDelegate = dg
        }
        
        xmlRoot = view
        for test in _layoutTests { dependency.pickLayout(view: view, passOrFail: test) }
        return view
    }
    
    public var _customView: CustomView!
    public var customView: CustomView { get { return _customView } set(value){ _customView = value } }
    public var _customViewDelegate: PongDelegate!
    public var customViewDelegate: PongDelegate { get { return _customViewDelegate } set(value) { _customViewDelegate = value } }
    
}
